bin/bump_version.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+# Retrieve the current version number and bump the versions of all required files by the specified amount and create a git version tag.
+# Based on https://gist.github.com/andyexeter/da932c9644d832e3be6706d20d539ff7
+
+# Usage: ./bin/bump_version.sh <major|minor|patch> - Increments the relevant version part by one.
+
+set -e
+
+
+function bump() {
+	echo -n "Updating $1..."
+	tmp_file=$(mktemp)
+	rm -f "$tmp_file"
+	sed -i "" "s/$2/$3/1w $tmp_file" $1
+	if [ -s "$tmp_file" ]; then
+		echo "Done"
+	else
+		echo "Nothing to change"
+	fi
+	rm -f "$tmp_file"
+}
+
+function confirm() {
+	read -r -p "$@ [Y/n]: " confirm
+
+	case "$confirm" in
+		[Nn][Oo]|[Nn])
+			echo "Aborting."
+			exit
+			;;
+	esac
+}
+
+
+if [ "$1" == "" ]; then
+	echo >&2 "No version type provided. Aborting."
+	exit 1
+fi
+
+if [ "$1" == "major" ] || [ "$1" == "minor" ] || [ "$1" == "patch" ]; then
+
+    # Get the current plugin version from the package.json
+    current_version=$(cat package.json | grep version | head -1  | awk -F: '{ print $2 }' | sed 's/[",]//g' | tr -d '[[:space:]]')
+
+	IFS='.' read -a version_parts <<< "$current_version"
+
+	major=${version_parts[0]}
+	minor=${version_parts[1]}
+	patch=${version_parts[2]}
+
+	case "$1" in
+		"major")
+			major=$((major + 1))
+			minor=0
+			patch=0
+			;;
+		"minor")
+			minor=$((minor + 1))
+			patch=0
+			;;
+		"patch")
+			patch=$((patch + 1))
+			;;
+	esac
+	new_version="$major.$minor.$patch"
+else
+	if [ "$2" == "" ]; then
+		echo >&2 "No 'to' version set. Aborting."
+		exit 1
+	fi
+	current_version="$1"
+	new_version="$2"
+fi
```

Comment:
I wonder if a design where this script takes the target version as input would work out to less code (basically kills this block) and more control.

Reply:
I was kind of aiming for an experience that would be more consistent with the `npm version` workflow and it felt like forcing the user to pass the current version as input would potentially leave more room for error.

Reply:
It would definitely leave more room for error. FWIW `npm version` supports either:

```
npm version [<newversion> | major | minor | patch
```